[QUARANTINE] TestSchedulerJobQueriesCount.test_process_dags_queries_count_02

Seems that this test fails randomly

Example here:

https://github.com/apache/airflow/runs/1239572585#step:6:421

Another example:

https://github.com/apache/airflow/pull/11453/checks?check_run_id=1240589774#step:6:414


```
_______ TestSchedulerJobQueriesCount.test_process_dags_queries_count_02 ________

a = (<tests.jobs.test_scheduler_job.TestSchedulerJobQueriesCount testMethod=test_process_dags_queries_count_02>,)

    @wraps(func)
    def standalone_func(*a):
>       return func(*(a + p.args), **p.kwargs)

/usr/local/lib/python3.6/site-packages/parameterized/parameterized.py:530: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/jobs/test_scheduler_job.py:3668: in test_process_dags_queries_count
    job._do_scheduling(session)
/usr/local/lib/python3.6/contextlib.py:88: in __exit__
    next(self.gen)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

expected_count = 14
>           raise AssertionError(message)
E           AssertionError: The expected number of db queries is 14. The current number is 23.
```



**Apache Airflow version**:


**Kubernetes version (if you are using kubernetes)** (use `kubectl version`):

**Environment**:

- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:

<!-- (please include exact error messages if you can) -->

**What you expected to happen**:

<!-- What do you think went wrong? -->

**How to reproduce it**:
<!---

As minimally and precisely as possible. Keep in mind we do not have access to your cluster or dags.

If you are using kubernetes, please attempt to recreate the issue using minikube or kind.

## Install minikube/kind

- Minikube https://minikube.sigs.k8s.io/docs/start/
- Kind https://kind.sigs.k8s.io/docs/user/quick-start/

If this is a UI bug, please provide a screenshot of the bug or a link to a youtube video of the bug in action

You can include images using the .md style of
![alt text](http://url/to/img.png)

To record a screencast, mac users can use QuickTime and then create an unlisted youtube video with the resulting .mov file.

--->


**Anything else we need to know**:

<!--

How often does this problem occur? Once? Every time etc?

Any relevant logs to include? Put them here in side a detail tag:
<details><summary>x.log</summary> lots of stuff </details>

-->

--- a/tests/jobs/test_scheduler_job.py
+++ b/tests/jobs/test_scheduler_job.py
@@ -3639,6 +3639,7 @@ class TestSchedulerJobQueriesCount(unittest.TestCase):
             # pylint: enable=bad-whitespace
         ]
     )
+    @pytest.mark.quarantined
     def test_process_dags_queries_count(
         self, expected_query_counts, dag_count, task_count, start_ago, schedule_interval, shape
     ):

